File: week1/py/F.py
import math

# Read the number of test cases 't' from the user.
t = int(input())

# Create a sieve to efficiently find prime numbers up to 100000.
sieve = [True] * 100000

# Use the Sieve of Eratosthenes algorithm to mark non-prime numbers.
for i in range(2, int(math.sqrt(100000)) + 1):
    if sieve[i] == True:
        for j in range(i * i, 100000, i):
            sieve[j] = False

# Iterate through each test case.
for _ in range(t):
    # Read two integers 'a' and 'b' from the user, representing the range [a, b].
    a, b = list(map(int, input().split()))
    
    # Initialize empty lists 'res' to store the twin prime pairs and 'count' to count them.
    res = []
    count = 0
    
    # Iterate through numbers in the range [a, b-1].
    for i in range(a, b - 1):
        # Check if both 'i' and 'i+2' are prime (twin prime pair).
        if sieve[i] == True and sieve[i + 2] == True:
            count += 1
            res.append((i, i + 2))
    
    # If no twin prime pairs are found, print "No."
    if count == 0:
        print("No")
    else:
        # Otherwise, print "Yes" and list the twin prime pairs.
        print("Yes")
        for i in res:
            print(i[0], i[1])

File: week1/cpp/F.cpp
#include <bits/stdc++.h>
using namespace std;
#define MAX 100000

int main() {
  int t;
  cin >> t; // Read the number of test cases.
  bool sieve[MAX]; // Create an array to represent the sieve of Eratosthenes.
  
  // Initialize the sieve array, assuming all numbers are prime initially.
  for (int i = 0; i < MAX; i++)
    sieve[i] = true;
  
  // Implement the Sieve of Eratosthenes algorithm to mark non-prime numbers.
  for (int i = 2; i * i <= MAX; i++) {
    if (sieve[i] == true) {
      for (int j = i * i; j <= MAX; j += i)
        sieve[j] = false;
    }
  }
  
  while (t--) {
    int l, r;
    cin >> l >> r; // Read the range [l, r] for the current test case.
    
    int count = 0; // Initialize a count variable to count twin prime pairs.
    vector<pair<int, int>> ans; // Initialize a vector to store twin prime pairs.
    
    // Iterate through numbers in the range [l, r-2] to find twin prime pairs.
    for (int i = l; i < r - 1; i++) {
      if (sieve[i] == true && sieve[i + 2] == true) {
        ans.push_back({i, i + 2}); // Add the twin prime pair to the vector.
        count++;
      }
    }
    
    // Check if twin prime pairs were found or not and print the result accordingly.
    if (!count)
      cout << "No\n";
    else {
      cout << "Yes\n";
      for (auto i : ans) {
        cout << i.first << " " << i.second << endl; // Print the twin prime pairs.
      }
    }
  }
}

